src/js/utils/ResourcesUtil.js

```diff
@@ -0,0 +1,127 @@
+import CompositeState from '../structs/CompositeState';
+import Node from '../structs/Node';
+import StringUtil from './StringUtil';
+
+const DefaultResourceTypes = {
+  cpus: {label: 'CPU', colorID: 0},
+  mem: {label: 'Memory', colorID: 6},
+  disk: {label: 'Disk', colorID: 3}
+};
+
+const usedColors = Object.keys(DefaultResourceTypes)
+.map(function (resource) {
+  return DefaultResourceTypes[resource].colorID;
+});
+const availableColors = Array(9).fill().map(function (value, index) {
+  return index;
+}).filter(function (value) {
+  return usedColors.indexOf(value) === -1;
+});
+
+function getAvailableColors() {
+  let colors = availableColors.slice(0);
+  // Concat the array twice. This is in case there's many resource.
+  // Unlikely that there will more than 4 anyways.
+  return colors.concat(colors);
+}
+
+const ResourcesUtil = {
+  getDefaultResources: function () {
+    return Object.keys(DefaultResourceTypes);
+  },
+
+  getAvailableResources: function (excludeList = []) {
+    let item = CompositeState.getNodesList().getItems()[0];
```

Comment:
I feel like we should use a safe accessor here instead of relying on it being `length > 0`. 

I know a cluster should technically always have nodes, but I can imagine some bad state occurring which causes reference errors.